Python cli

KritsanaMike opened the pull request:
add cli for get information from i2c and swarm tile

KritsanaMike committed:
add comment

KritsanaMike committed:
up data cli to get i2c information and Swarm information

antonyharfield commented on cli/commands/swarm.py:
```diff
@@ -5,15 +5,52 @@
+@app.command()
+def gps(): # TODO Get GPS Fix Quality (Example return $GS 109,214,9,0,G3*46)
+    device = Device()
+    result = execute(device, '$GS @*74') 
```
Let's calculate the checksum automatically. This line should be:

```
result = execute(device, 'GS @')
```

The `$` and the `*XX` should be added by the execute command.

antonyharfield commented on cli/commands/adb_guardian.py:
```diff
@@ -18,7 +18,7 @@ def start():
     # print("i2c",i2c,type(i2c))
 
-    # ECHO Input Voltage 
+    # ECHO Input Voltage -
```
?

antonyharfield commented on cli/commands/i2c.py:
```diff
@@ -0,0 +1,76 @@
+
+
+@app.command()
+def status(): # TODO Get i2c status 
```
Remove `TODO`

KritsanaMike committed:
fix code for easy use and maintenance

- separate code for use shell command 2 core file
- make ui.py to display ui .exe file

antonyharfield commented on cli/README.md:
```diff
@@ -5,7 +5,17 @@
+- `i2c system-info` Get the  system information (Voltage)
+- `i2c bat-info` Get the battery information (Voltage, Current, battery percentage)
+
+- `ui` Show ui .exe file 
```
The UI is not related to `.exe` file.

antonyharfield commented on cli/commands/common/swarm_command.py:
```diff
@@ -0,0 +1,63 @@
+    if int(checksum, base=16) == calculated_checksum:
+        return nmeadata
+    else:
+        raise ValueError("The NMEA data does not match it's checksum")
```
Not being used?

antonyharfield commented on cli/commands/swarm.py:
```diff
@@ -1,19 +1,46 @@
+    sw_status = sw_status[0].split(":")
+    sw_status = sw_status[1].split("\r")
+
+    if sw_status[0] == "0011010-1":
```
Refactor: This logic is needed in UI and CLI.

antonyharfield commented on cli/commands/common/swarm_command.py:
```diff
@@ -0,0 +1,63 @@
+    startchars = "!$"
+    for c in startchars:
+        i = msg.find(c)
+        if i >= 0: break
```
This is confusing... it adds the $ and then try to remove it in the loop.

antonyharfield commented on cli/commands/common/swarm_command.py:
```diff
@@ -0,0 +1,63 @@
+        i = msg.find(c)
+        if i >= 0: break
+    else:
+        return (False, None, None)
```
I don't see where (False, None, None) is needed.

antonyharfield reviewed (commented):
Good progress!

There are a few bits of logic in the UI and CLI that I think can be refactored to 1 place. E.g. when getting a message from swarm we can use the same code to remove the checksum and parse the message.

KritsanaMike committed:
update code

- clean data from read command before send out
- clear bug on UI
- add command to make . exe program

Tooseriuz commented on cli/commands/common/i2c_command.py:
```diff
@@ -0,0 +1,56 @@
+    bat_v = int(bat_v, 16)*0.192264
+    return float(bat_v)
+
+def battery_cuurent(device):
```
`cuurent` to `current`

Tooseriuz commented on cli/commands/common/swarm_command.py:
```diff
@@ -0,0 +1,83 @@
+    shell_command = f'echo -n \'{command}\\r\' > /dev/ttyMT1 | /system/xbin/busybox timeout 1 /system/bin/cat < /dev/ttyMT1'
+    response = device.shell(shell_command)
+    lines = response.split("\n")
+    # Validate checksum and remove $ *
+    validated_lines = [validate_checksum(line) for line in lines] 
```
I don't think we need to validate checksum on result of the command tho

KritsanaMike replied:
for this and help me to write the code. and I think we validate checksum because the value from swarm will include $ and *. if we put into check we will get real value

antonyharfield replied:
@Tooseriuz Validate checksum helps be confident that a corrupt message was not sent.

Tooseriuz reviewed (changes requested):
not sure why we need to validate checksum when getting result back from command

KritsanaMike committed:
fix name function and fix read me description

antonyharfield commented on cli/commands/common/i2c_command.py:
```diff
@@ -0,0 +1,56 @@
+    if i2c_status == "0x0001":
+        return True
+    else:
+        return False
```
`return i2c_status == "0x0001"`

antonyharfield commented on cli/commands/common/i2c_command.py:
```diff
@@ -0,0 +1,56 @@
+    if response != None:
+        return response
+    else:
+        return None
```
Ewww... the 4 lines are the same as `return response`

antonyharfield commented on cli/commands/common/i2c_command.py:
```diff
@@ -0,0 +1,56 @@
+
+def input_voltage(device):
+    result = execute(device, '0x3b') 
+    input_v = result[0:6]
```
Avoid duplicating code.

You do this on every command, so it can be moved into `execute`.

antonyharfield commented on cli/commands/common/i2c_command.py:
```diff
@@ -0,0 +1,56 @@
+def input_voltage(device):
+    result = execute(device, '0x3b') 
+    input_v = result[0:6]
+    input_v = int(input_v, 16)*1.648
```
Also do this on every command `int(x, 16)`. Move it inside.

antonyharfield commented on cli/commands/common/i2c_command.py:
```diff
@@ -0,0 +1,56 @@
+def input_current(device):
+    result = execute(device, '0x3e') 
+    input_c = result[0:6]
+    input_c = int(input_c, 16)*(0.00146487 /0.005)
```
Inconsistent spacing.

antonyharfield commented on cli/commands/common/swarm_command.py:
```diff
@@ -0,0 +1,83 @@
+    for v in validated_lines:
+        if v.startswith(prefix):
+            value = v
+    return value
```
Same as:
```
   for v in validated_lines:
         if v.startswith(prefix):
             return v
   return None
```

antonyharfield replied:
Avoid creating unnecessary variables.

antonyharfield commented on cli/commands/common/swarm_command.py:
```diff
@@ -0,0 +1,83 @@
+        if v.startswith(prefix):
+            value = v
+    return value
+    # return [line for line in validated_lines if line.startswith(prefix)][0]
```
Avoid committing commented code.

antonyharfield commented on cli/commands/common/swarm_command.py:
```diff
@@ -0,0 +1,83 @@
+    if swarm_fw != None:
+        return str(swarm_fw.split(" ")[1])
+    else:
+        return None
```
When using `return` in `if`, you can:
```
if swarm_fw != None:
   return str(swarm_fw.split(" ")[1])
return None
```

antonyharfield commented on cli/commands/common/swarm_command.py:
```diff
@@ -0,0 +1,83 @@
+        else:
+            raise ValueError("The NMEA data does not match it's checksum")
+    else:
+        return "None" # startswith method cannot check None value
```
Ugly!

If your code to detect `None` before calling `startswith`.

antonyharfield commented on cli/commands/i2c.py:
```diff
@@ -0,0 +1,48 @@
+    i2c_status = i2cStatus(device)
+    if i2c_status == True:
+        typer.echo("i2c status: " + typer.style("i2c is ON", fg=typer.colors.GREEN, bold=True))
+    else :
```
Inconsistent space before :

antonyharfield commented on cli/commands/i2c.py:
```diff
@@ -0,0 +1,48 @@
+    device = Device()
+    sys_v = system_voltage(device)
+    sys_v = float("{:.2f}".format(sys_v))
+    typer.echo("System voltage: " + typer.style(sys_v, fg=typer.colors.GREEN, bold=True))
```
float => string => float 😱

Suggest:
```
typer.echo("System voltage: " + typer.style("{:.2f}".format(sys_v), fg=typer.colors.GREEN, bold=True))
```

KritsanaMike committed:
fix code follow reviewer

- Avoid duplicating code.
- Inconsistent spacing.

antonyharfield committed:
Update README.md

KritsanaMike committed:
fix return

KritsanaMike committed:
fix return value

KritsanaMike committed:
fix return validate checksum
